get_active_hooks.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+"""
+    Report on Service & Web hooks for organization.
+"""
+import argparse
+import client
+import logging
+import urlparse
+import yaml
+
+"""
+    Lore:   swapping hook.test for hook.ping will cause repetition of the
+            actions.  In particular, a number of repos post to IRC channels
+            and/or bugs on commits, so expect comments to that effect.
+"""
+
+logger = logging.getLogger(__name__)
+
+
+def report_hooks(gh, org, active_only, unique_only, do_ping, yaml_out):
+    org_handle = gh.organization(org)
+    org_struct = org_handle.to_json()
+    repo_list = []
+    org_struct['repo_list'] = repo_list
+    unique_hooks = set()
+    msg = "Active" if active_only else "All"
+    for repo in org_handle.iter_repos():
+        repo_struct = repo.to_json()
+        hook_list = []
+        repo_struct['hook_list'] = hook_list
+        repo_list.append(repo_struct)
+        repo_hooks = set()
+        ping_attempts = ping_fails = 0
+        for hook in repo.iter_hooks():
+            hook_struct = hook.to_json()
+            hook_list.append(hook_struct)
+            # if hook.name == "web", then this is a web hook, and there can be
+            # several per repo. The unique part is the hook.config['url'], which
+            # may contain sensitive info (including basic login data), so just
+            # grab scheme, hostname, and port.
+            if hook.name != "web":
+                name = hook.name
+            else:
+                url = hook.config['url']
+                parts = urlparse.urlparse(url)
+                # port can be None, which prints funny, but is good enough for
+                # identification.
+                name = "%s://%s:%s" % (parts.scheme, parts.hostname, parts.port)
+            if hook.active or not active_only:
+                repo_hooks.add(name)
+            if do_ping and hook.active:
+                ping_attempts += 1
+                if not hook.ping():
```

Comment:
Side effects in if statements are frowned upon in Python (which is why things like `if errors++:`` do not work by design).